src/middleware/authJWT.js

```diff
@@ -0,0 +1,38 @@
+const jwt = require("jsonwebtoken");
+const userList = require("../user.json");
+
+//To validate given token with stored secret
+const verifyToken = (req, res, next) => {
+  if (
+    req.headers &&
+    req.headers.authorization &&
+    req.headers.authorization.split(" ")[0] === "JWT"
+  ) {
+    jwt.verify(
+      req.headers.authorization.split(" ")[1],
+      process.env.API_SECRET,
+      function (err, decode) {
+        if (err) {
+          req.user = undefined;
+          next();
```

Comment:
in cases of failures, the middleware should return a proper error message with the relevant status code.